using_restricted_array.rb

```diff
@@ -31,20 +31,26 @@ def search(array, length, value_to_find)
   length.times do |i|
     while array[i] != value_to_find
```

Comment:
You don't really need this while loop. The outer times loop and the inner conditional statement should be enough.